### PR TITLE
Improve sq device mapping and enhance ut

### PR DESCRIPTION
**Pull request by changwangss**
## Type of Change

- the example input of jit trace is (input_ids, past-kv, attention_mask) for text-generation task, the device mapping will cause error.
- enhance sq ut for text-generation task.
  - add the modeling_gptj.py due to ipex 2.0 trace gptj failed. it will remove when ipex/pt 2.1 release.

## Description

detail description

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

**Review by yintong-lu — APPROVED** [review]

File: neural_compressor/adaptor/torch_utils/smooth_quant.py [hunk 1]
```diff
@@ -57,7 +57,10 @@ def move_input_to_device(input, device=torch.device('cpu')):
 
 ##TODO potential bug, data type
 def forward_wrapper(model, input, device=torch.device('cpu')):
-    input = move_input_to_device(input, device)
+    try:
+        input = move_input_to_device(input, device)
```
Comment (wenhuach21) [inline neural_compressor/adaptor/torch_utils/smooth_quant.py:61]: @yintong-lu  could you help support recursive solution

Comment (yintong-lu) [reply neural_compressor/adaptor/torch_utils/smooth_quant.py:61]: yes, I will create a seperate pr for it.

File: neural_compressor/adaptor/torch_utils/smooth_quant.py [hunk 2]
```diff
@@ -1205,6 +1208,8 @@ def trace(self, model, dummy_input):
         optimize_numerics = False
         if hasattr(model, "device"):
             orig_device = model.device
+            if hasattr(model, "type"):
```
Comment (xin3he) [inline neural_compressor/adaptor/torch_utils/smooth_quant.py:1211]: model.device, 'type'? BTW, how to handle the index if cuda:0/1

**Review by xin3he — CHANGES_REQUESTED** [review]

File: neural_compressor/adaptor/torch_utils/smooth_quant.py (hunk 2 above)
Comment (changwangss) [reply neural_compressor/adaptor/torch_utils/smooth_quant.py:1211]: orig_device will be "cuda:0/1", if the origin_device != cpu, the tensor will convert to cpu tensor.

**Review by xin3he — APPROVED** [review]

**Pull request merged**